decisions/0005-file-format.md

```diff
@@ -0,0 +1,134 @@
+# File Format
+
+- Status: **Accepted**
+- Deciders: Vlad Ionica, Dan Nixon, Adam Washington, Tristan Youngs
+- Date:  11-03-2022
+
+## Context of the problem
+
+The project to date has used a homegrown file format.  While fit for
+purpose, it does add overhead for managing the parsing and writing.
+Additionally, as it is a custom format, there exists no tooling for
+managing and editing these files.
+
+Moving to an industry standard file format would enable us to farm out
+parsing and serialisation to a library while also unlocking a
+community of tooling and support.
+
+## Decision Drivers
+
+- The file format must be able to represent the entire contents of a Dissolve input file.
+- A high quality, stable library for the file format must be available.
+- Any library should prefer modern C++ over pointer passing semantics.
+- The format must be amenable to human editing and adding comments.
+- An error in hand editing a file should lead to a failed parse and not an incorrect input.
+- Will comments persist after editing in Dissolve?
+- Can common text editors provide highlighting support for this format
+- Tools should be available for querying and transforming these files.
+- It should be possible to update the Dissolve schema without invalidating all files in the old format.
+- Is it possible to simultaneously support multiple schema versions?
+- Is the format reasonably efficient.
+
+## Considered Options
+
+1. XML
+2. JSON
+3. YAML
+4. TOML
+5. Dhall
+
+## Decision Outcome
+
+Chosen option: TOML.  It provided better editing support than XML or
+JSON, while having a less ambiguous syntax than YAML (forcing broken
+files into failed parses instead of faulty simulations).  Dhall was
+ruled out almost immediately due to poor library support.
+
+Toml11 was chosen for the parsing library.  The API was intuitive and
+was similar enough to the API of the top YAML library that it would be
+possible to quickly switch out implementations, if a problem with the
+library or format was found.
+
+## Positive Condequences
+
+- The `LineParser` class and most of our IO code can be eliminated,
```

Comment:
Suggest that this should be rephrased as reducing our dependence on `LineParser` rather than enabling its removal. We still depend on it for all output file writing (restart files could be moved to TOML, but simulation data output cannot).